internal/scalibrplugin/extractors_parser.go

```diff
@@ -2,17 +2,72 @@
 package scalibrplugin
 
 import (
+	"maps"
+	"slices"
+
 	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/java/pomxml"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/java/pomxmlnet"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/javascript/packagejson"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/python/requirements"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/python/requirementsnet"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/rust/cargoauditable"
+	"github.com/google/osv-scalibr/extractor/filesystem/list"
+	"github.com/google/osv-scalibr/extractor/filesystem/secrets"
 	"github.com/google/osv-scanner/v2/internal/builders"
-	"github.com/google/osv-scanner/v2/internal/scalibrextract"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/filesystem/vendored"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/language/java/pomxmlenhanceable"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/language/javascript/nodemodules"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/language/python/requirementsenhancable"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/vcs/gitrepo"
 )
 
-// todo: rename this to be clearer now we've got `detectorPresets`
-var presets = map[string][]string{
-	"sbom":      scalibrextract.ExtractorsSBOMs,
-	"lockfile":  scalibrextract.ExtractorsLockfiles,
-	"directory": scalibrextract.ExtractorsDirectories,
-	"artifact":  scalibrextract.ExtractorsArtifacts,
+var extractorPresets = map[string]list.InitMap{
+	"sbom": list.SBOM,
+	"lockfile": concat(
+		without(list.SourceCode, []string{
+			pomxml.Name, pomxmlnet.Name,
+			requirements.Name, requirementsnet.Name,
+			// this is actually a detector
+			secrets.Name,
```

Comment:
@another-rex @erikvarga if you agree this should be a detector, I can have a crack at making that move (and that would also mean we'd have a simple well-tested detector for our e2e tests as a nice bonus)

Reply:
for reference, this is why I don't think we should have it enabled by default right now

<img width="1546" height="1657" alt="image" src="https://github.com/user-attachments/assets/555028cd-23b3-4e22-b6c7-023405ba74c7" />